pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -132,16 +132,16 @@ func (state PipelineRunState) IsBeforeFirstTaskRun() bool {
 func (state PipelineRunState) AdjustStartTime(unadjustedStartTime *metav1.Time) *metav1.Time {
 	adjustedStartTime := unadjustedStartTime
 	for _, rpt := range state {
-		if rpt.TaskRun == nil {
-			if rpt.RunObject != nil {
-				creationTime := rpt.RunObject.GetObjectMeta().GetCreationTimestamp()
+		if len(rpt.TaskRuns) == 0 {
+			if len(rpt.RunObjects) > 0 {
```

Comment:
I think here it makes more sense to loop over run objects and get the earliest of their start times.